src/test/resources/BulkCopyCSVTestInputDelimiterEscape.csv

```diff
@@ -0,0 +1,5 @@
+1,"test", " test""", "no,split" , testNoQuote

```

Comment:
Cases such as ` " test"""` and ` "test""test"` are actually invalid cases. RFC 4180 says that spaces need to be insides double quotes if double quotes are provided, so spaces cannot come before or after the double quotes (refer to https://stackoverflow.com/questions/4863852/space-before-quote-in-csv-field for some explanation on this too). 

Currently the code allows this case and simply ignores the spaces, but I think we should not allow these cases (and throw an exception saying that there is an invalid row), then change this CSV file to not test for these invalid cases.

Reply:
You are right that strictly based on the following statement, there should be no spaces before and after enclosing quotes. 

> Spaces are considered part of a field and should not be ignored.

But I found this [WIKI page](https://en.wikipedia.org/wiki/Comma-separated_values#RFC_4180_standard):

> According to RFC 4180, spaces outside quotes in a field are not allowed; however, the RFC also says that "Spaces are considered part of a field and should not be ignored." and "Implementors should 'be conservative in what you do, be liberal in what you accept from others' (RFC 793, section 2.10) when processing CSV files."

> 1997, "Ford" ,E350

I also did some research online and found multiple issues where users run into difficulties because of this limitation in the specs. And I don't see what benefit we would get from throwing an exception instead of ignoring it, so I chose to give flexibility to applications that won't impact the functionality of the feature.

Reply:
Ok, I think that's a valid option too. We can leave the code as it is, but we might want to add a short comment saying that the code will ignore spaces that come before and after double quotes, since it might be relevant to some users. I'll resolve my other comment related to this as well.

Reply:
I added a comment to the Javadoc of `setEscapeColumnDelimitersCSV`.